src/modules/attempts/attempts.service.ts

```diff
@@ -113,6 +117,514 @@ export class AttemptsService {
     return savedAttempt;
   }
 
+
+  /**
+   * Retrieves a test attempt with all associated data for resuming an in-progress test.
+   * This method orchestrates the retrieval and transformation of attempt data including
+   * test details, questions, user answers, and progress metrics.
+   * 
+   * @param attemptId - The unique identifier of the attempt to retrieve
+   * @param authContext - Authentication context containing user and organization details
+   */
+  async getAttempt(attemptId: string, authContext: AuthContext): Promise<any> {
+    // Step 1: Validate and retrieve the attempt
+    const attempt = await this.validateAndRetrieveAttempt(attemptId, authContext);
```

Comment:
@vinayak-patil when attempt is resumed you should update the updatedAt in userAttempts ?